va/va.h

```diff
@@ -3777,6 +3777,22 @@ VAStatus vaQuerySurfaceError(
     void **error_info
 );
 
+/**
+ * This function blocks during specified timeout (in nanoseconds) until
+ * all pending operations associated with given buf_id have been completed.
+ * If timeout is zero, the function returns immediately.
+ * A common example of usage is synchronization of output encoded buffers, output statistic buffers.
+ * Return values:
+ * VA_STATUS_SUCCESS - all pending operations completed successfully;
+ * VA_STATUS_ERROR_HW_BUSY - operations are still in progress;
+ * the other error statuses indicating that operations failed;
+ */
+VAStatus vaSyncBuffer(
+    VADisplay dpy,
+    VABufferID buf_id,
+    uint64_t timeout_ns
+);
```

Comment:
Need to think on how to get status reports (like encoded buffer size, QP, num passes, etc). Generally we can do something similar to https://github.com/intel/libva/blob/82122961145d4be616400cc99485867adfddd971/va/va.h#L3419 but there are just 4 bits left. We need something extensible.